sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -537,10 +603,135 @@ def peek_messages(self, max_message_count=1, **kwargs):
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ReceivedMessage")
+        self._check_message_alive(message, MESSAGE_DEFER)
+        self._settle_message_with_retry(message, MESSAGE_DEFER)
+        message._settled = True  # pylint: disable=protected-access
+
+    def dead_letter_message(self, message, reason=None, error_description=None):
+        """Move the message to the Dead Letter queue.
+
+        The Dead Letter queue is a sub-queue that can be
+        used to store messages that failed to process correctly, or otherwise require further inspection
+        or processing. The queue can also be configured to send expired messages to the Dead Letter queue.
+
+        :param message: The received message to be dead-lettered.
+        :type message: ~azure.servicebus.ReceivedMessage
+        :param Optional[str] reason: The reason for dead-lettering the message.
+        :param Optional[str] error_description: The detailed error description for dead-lettering the message.
+        :rtype: None
+        :raises: ~azure.servicebus.exceptions.MessageAlreadySettled if the message has been settled.
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ReceivedMessage")
+        self._check_message_alive(message, MESSAGE_DEAD_LETTER)
+        self._settle_message_with_retry(
+            message,
+            MESSAGE_DEAD_LETTER,
+            dead_letter_reason=reason,
+            dead_letter_error_description=error_description
+        )
+        message._settled = True  # pylint: disable=protected-access
+
+    def renew_message_lock(self, message, **kwargs):
+        # type: (ServiceBusReceivedMessage, Any) -> datetime.datetime
+        # pylint: disable=protected-access,no-member
+        """Renew the message lock.
+
+        This will maintain the lock on the message to ensure it is not returned to the queue
+        to be reprocessed.
+
+        In order to complete (or otherwise settle) the message, the lock must be maintained,
+        and cannot already have expired; an expired lock cannot be renewed.
+
+        Messages received via ReceiveAndDelete mode are not locked, and therefore cannot be renewed.
+        This operation is only available for non-sessionful messages as well.
+
+        :keyword float timeout: The total operation timeout in seconds including all the retries. The value must be
```

Comment:
Need to document the `message` parameter

Reply:
good catch, added